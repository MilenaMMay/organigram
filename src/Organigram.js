import React, { Component } from 'react'
import PropTypes from 'prop-types'

import organigramStyles from './Organigram.styles'

export default class Organigram extends Component {
  static propTypes = {
    text: PropTypes.string,
    data: PropTypes.object,
  }

  render() {
    const {
      text,
      data,
    } = this.props

    return (
      <div>
        {data.views.map((view) =>
          <div style={organigramStyles} key={view.name}>
            {view.name}
          </div>
          )}
      </div>
    )
  }
}
