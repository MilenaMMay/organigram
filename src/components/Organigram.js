import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Tabs from '@material-ui/core/Tabs'
import Tab from '@material-ui/core/Tab'
import View from './View'

export default class Organigram extends Component {
  static propTypes = {
    data: PropTypes.object,
  }

  constructor(props) {
    super(props);
    this.state = {
      data: props.data,
      selectedView: props.data.views[0]
    }
  }

  render() {
    return (
      <div>
        <Tabs
          value={this.state.selectedView.name}
          onChange={this.handleChange}
          indicatorColor="primary"
          textColor="primary"
          centered
        >
          {this.state.data.views.map((view) =>
            <Tab
              label={view.name}
              value={view.name}
              key={view.name}
              onClick={() => this.setState({selectedView: view})}
            />
          )}
        </Tabs>
        { this.state.selectedView ? <View view={this.state.selectedView}></View> : null }
      </div>
    )
  }
}
