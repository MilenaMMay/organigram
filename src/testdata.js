export const person1 = { firstname: 'Person' , lastname: 'One', role: 'no1' }
export const person2 = { firstname: 'Person' , lastname: 'Two', role: 'no2' }
export const person3 = { firstname: 'Person' , lastname: 'Three', role: 'no3' }
export const person4 = { firstname: 'Person' , lastname: 'Four', role: 'no4' }
export const person5 = { firstname: 'Person' , lastname: 'Five', role: 'no5' }
export const person6 = { firstname: 'Person' , lastname: 'Six', role: 'no6' }

export const group1 = { groupname: 'Group 1', members: [ person4, person6 ] }

export const smalltree = { name: 'Small tree' , parent: person1, children: [ person2, person3, group1 ] }

export const group2 = { groupname: 'Group 2', members: [ smalltree ] }

export const bigtree = { name: 'Big tree' , parent: group2, children: [ smalltree, person5 ] }

// TODO Rename trees because it may also contain groups
export const view1 = { name: 'View One' , trees: [ smalltree, bigtree, group1 ] }
export const view2 = { name: 'View Two' , trees: [ bigtree ] }

export const data = { views: [ view1, view2 ] }
