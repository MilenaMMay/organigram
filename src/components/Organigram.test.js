import React from 'react';
import Organigram from './Organigram'
import renderer from 'react-test-renderer';
import { data } from '../testdata';

test('Organigram renders views', () => {
  const component = renderer.create(
    <Organigram data={data}></Organigram>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
