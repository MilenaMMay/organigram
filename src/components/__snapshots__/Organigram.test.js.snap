// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organigram renders views 1`] = `
<div>
  <div
    className="MuiTabs-root-1"
  >
    <div
      className="MuiTabs-flexContainer-2"
    >
      <div
        className="MuiTabs-scroller-4 MuiTabs-fixed-5"
        onScroll={[Function]}
        role="tablist"
        style={
          Object {
            "marginBottom": 0,
          }
        }
      >
        <div
          className="MuiTabs-flexContainer-2 MuiTabs-centered-3"
        >
          <button
            aria-selected={true}
            className="MuiButtonBase-root-22 MuiTab-root-10 MuiTab-textColorPrimary-13 MuiTab-selected-15"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiTab-wrapper-18"
            >
              <span
                className="MuiTab-labelContainer-19"
              >
                <span
                  className="MuiTab-label-20"
                >
                  View One
                </span>
              </span>
            </span>
            <span
              className="MuiTouchRipple-root-220"
            />
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root-22 MuiTab-root-10 MuiTab-textColorPrimary-13"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiTab-wrapper-18"
            >
              <span
                className="MuiTab-labelContainer-19"
              >
                <span
                  className="MuiTab-label-20"
                >
                  View Two
                </span>
              </span>
            </span>
            <span
              className="MuiTouchRipple-root-220"
            />
          </button>
        </div>
        <span
          className="MuiPrivateTabIndicator-root-25 MuiPrivateTabIndicator-colorPrimary-26 MuiTabs-indicator-9"
          style={
            Object {
              "left": 0,
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>
  <div>
    <div
      className="MuiPaper-root-32 MuiPaper-elevation1-35 MuiPaper-rounded-33 MuiExpansionPanel-root-28 MuiExpansionPanel-rounded-29"
    >
      <div
        aria-expanded={false}
        className="MuiButtonBase-root-22 MuiExpansionPanelSummary-root-59"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="MuiExpansionPanelSummary-content-63"
        >
          <h4
            className="MuiTypography-root-65 MuiTypography-h4-80 MuiTypography-colorTextSecondary-98"
          >
            Small tree
          </h4>
        </div>
        <div
          aria-hidden="true"
          className="MuiButtonBase-root-22 MuiIconButton-root-101 MuiExpansionPanelSummary-expandIcon-64"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={-1}
        >
          <span
            className="MuiIconButton-label-106"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-107"
              color={undefined}
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-220"
          />
        </div>
      </div>
      <div
        aria-hidden="true"
        className="MuiCollapse-container-116"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper-118"
        >
          <div
            className="MuiCollapse-wrapperInner-119"
          >
            <div
              className="MuiExpansionPanelDetails-root-120"
            >
              <div
                className="MuiPaper-root-32 MuiPaper-elevation1-35 MuiPaper-rounded-33 MuiCard-root-121"
              >
                <div
                  className="MuiCardContent-root-122"
                >
                  <h5
                    className="MuiTypography-root-65 MuiTypography-h5-81"
                  >
                    Person
                     
                    One
                  </h5>
                  <p
                    className="MuiTypography-root-65 MuiTypography-body1-74 MuiTypography-colorTextSecondary-98"
                  >
                    no1
                  </p>
                </div>
              </div>
              <div
                className="MuiGrid-container-123 MuiGrid-align-items-xs-center-131 MuiGrid-justify-xs-space-evenly-144"
              >
                <div
                  className="MuiPaper-root-32 MuiPaper-elevation1-35 MuiPaper-rounded-33 MuiCard-root-121"
                >
                  <div
                    className="MuiCardContent-root-122"
                  >
                    <h5
                      className="MuiTypography-root-65 MuiTypography-h5-81"
                    >
                      Person
                       
                      Two
                    </h5>
                    <p
                      className="MuiTypography-root-65 MuiTypography-body1-74 MuiTypography-colorTextSecondary-98"
                    >
                      no2
                    </p>
                  </div>
                </div>
                <div
                  className="MuiPaper-root-32 MuiPaper-elevation1-35 MuiPaper-rounded-33 MuiCard-root-121"
                >
                  <div
                    className="MuiCardContent-root-122"
                  >
                    <h5
                      className="MuiTypography-root-65 MuiTypography-h5-81"
                    >
                      Person
                       
                      Three
                    </h5>
                    <p
                      className="MuiTypography-root-65 MuiTypography-body1-74 MuiTypography-colorTextSecondary-98"
                    >
                      no3
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root-32 MuiPaper-elevation1-35 MuiPaper-rounded-33 MuiExpansionPanel-root-28 MuiExpansionPanel-rounded-29"
    >
      <div
        aria-expanded={false}
        className="MuiButtonBase-root-22 MuiExpansionPanelSummary-root-59"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="MuiExpansionPanelSummary-content-63"
        >
          <h4
            className="MuiTypography-root-65 MuiTypography-h4-80 MuiTypography-colorTextSecondary-98"
          >
            Big tree
          </h4>
        </div>
        <div
          aria-hidden="true"
          className="MuiButtonBase-root-22 MuiIconButton-root-101 MuiExpansionPanelSummary-expandIcon-64"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={-1}
        >
          <span
            className="MuiIconButton-label-106"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-107"
              color={undefined}
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-220"
          />
        </div>
      </div>
      <div
        aria-hidden="true"
        className="MuiCollapse-container-116"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper-118"
        >
          <div
            className="MuiCollapse-wrapperInner-119"
          >
            <div
              className="MuiExpansionPanelDetails-root-120"
            >
              <div
                className="MuiPaper-root-32 MuiPaper-elevation1-35 MuiPaper-rounded-33 MuiCard-root-121"
              >
                <div
                  className="MuiCardContent-root-122"
                >
                  <h5
                    className="MuiTypography-root-65 MuiTypography-h5-81"
                  >
                    Person
                     
                    Four
                  </h5>
                  <p
                    className="MuiTypography-root-65 MuiTypography-body1-74 MuiTypography-colorTextSecondary-98"
                  >
                    no4
                  </p>
                </div>
              </div>
              <div
                className="MuiGrid-container-123 MuiGrid-align-items-xs-center-131 MuiGrid-justify-xs-space-evenly-144"
              >
                <div
                  className="MuiPaper-root-32 MuiPaper-elevation1-35 MuiPaper-rounded-33 MuiExpansionPanel-root-28 MuiExpansionPanel-rounded-29"
                >
                  <div
                    aria-expanded={false}
                    className="MuiButtonBase-root-22 MuiExpansionPanelSummary-root-59"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex="0"
                  >
                    <div
                      className="MuiExpansionPanelSummary-content-63"
                    >
                      <h4
                        className="MuiTypography-root-65 MuiTypography-h4-80 MuiTypography-colorTextSecondary-98"
                      >
                        Small tree
                      </h4>
                    </div>
                    <div
                      aria-hidden="true"
                      className="MuiButtonBase-root-22 MuiIconButton-root-101 MuiExpansionPanelSummary-expandIcon-64"
                      onBlur={[Function]}
                      onContextMenu={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={-1}
                    >
                      <span
                        className="MuiIconButton-label-106"
                      >
                        <svg
                          aria-hidden="true"
                          className="MuiSvgIcon-root-107"
                          color={undefined}
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                          />
                          <path
                            d="M0 0h24v24H0z"
                            fill="none"
                          />
                        </svg>
                      </span>
                      <span
                        className="MuiTouchRipple-root-220"
                      />
                    </div>
                  </div>
                  <div
                    aria-hidden="true"
                    className="MuiCollapse-container-116"
                    style={
                      Object {
                        "minHeight": "0px",
                      }
                    }
                  >
                    <div
                      className="MuiCollapse-wrapper-118"
                    >
                      <div
                        className="MuiCollapse-wrapperInner-119"
                      >
                        <div
                          className="MuiExpansionPanelDetails-root-120"
                        >
                          <div
                            className="MuiPaper-root-32 MuiPaper-elevation1-35 MuiPaper-rounded-33 MuiCard-root-121"
                          >
                            <div
                              className="MuiCardContent-root-122"
                            >
                              <h5
                                className="MuiTypography-root-65 MuiTypography-h5-81"
                              >
                                Person
                                 
                                One
                              </h5>
                              <p
                                className="MuiTypography-root-65 MuiTypography-body1-74 MuiTypography-colorTextSecondary-98"
                              >
                                no1
                              </p>
                            </div>
                          </div>
                          <div
                            className="MuiGrid-container-123 MuiGrid-align-items-xs-center-131 MuiGrid-justify-xs-space-evenly-144"
                          >
                            <div
                              className="MuiPaper-root-32 MuiPaper-elevation1-35 MuiPaper-rounded-33 MuiCard-root-121"
                            >
                              <div
                                className="MuiCardContent-root-122"
                              >
                                <h5
                                  className="MuiTypography-root-65 MuiTypography-h5-81"
                                >
                                  Person
                                   
                                  Two
                                </h5>
                                <p
                                  className="MuiTypography-root-65 MuiTypography-body1-74 MuiTypography-colorTextSecondary-98"
                                >
                                  no2
                                </p>
                              </div>
                            </div>
                            <div
                              className="MuiPaper-root-32 MuiPaper-elevation1-35 MuiPaper-rounded-33 MuiCard-root-121"
                            >
                              <div
                                className="MuiCardContent-root-122"
                              >
                                <h5
                                  className="MuiTypography-root-65 MuiTypography-h5-81"
                                >
                                  Person
                                   
                                  Three
                                </h5>
                                <p
                                  className="MuiTypography-root-65 MuiTypography-body1-74 MuiTypography-colorTextSecondary-98"
                                >
                                  no3
                                </p>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="MuiPaper-root-32 MuiPaper-elevation1-35 MuiPaper-rounded-33 MuiCard-root-121"
                >
                  <div
                    className="MuiCardContent-root-122"
                  >
                    <h5
                      className="MuiTypography-root-65 MuiTypography-h5-81"
                    >
                      Person
                       
                      Five
                    </h5>
                    <p
                      className="MuiTypography-root-65 MuiTypography-body1-74 MuiTypography-colorTextSecondary-98"
                    >
                      no5
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
